WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/home/usecases/GetPublishButtonLabelUseCase.kt

```diff
@@ -0,0 +1,30 @@
+package org.wordpress.android.ui.posts.prepublishing.home.usecases
+
+import android.text.TextUtils
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.post.PostStatus
+import org.wordpress.android.ui.posts.EditPostRepository
+import org.wordpress.android.ui.posts.PostUtilsWrapper
+import javax.inject.Inject
+
+typealias StringResourceId = Int
+
+class GetPublishButtonLabelUseCase @Inject constructor(private val postUtilsWrapper: PostUtilsWrapper) {
```

Comment:
Hey @malinajirka 
This logic was copied from the logic that is used in `PostSettingsUtil` I had mentioned in the PR. However, all these claims you are making here are indeed correct and I have another branch that contains changes that eliminate most of this logic. I was just keeping this PR here as I figured out a way to branch from this without making the other PR too large.

Reply:
So yes, it's the same logic that's being modified here https://github.com/wordpress-mobile/WordPress-Android/pull/11887 .  honestly, I was a bit concerned about how to make the changes there as I wasn't sure of the priority/precedence that conditions should take depending on time/`PostStatus` so your comment is very timely.

Reply:
> How come isPublishDateInTheFuture can be true but the post not be in SCHEDULED state?

@malinajirka so I am here doing some thinking about this and looking at the code, in particular, this block here

https://github.com/wordpress-mobile/WordPress-Android/blob/9c5cd8296da5fee077cb812a066a9da07ea90d99/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt#L121

So as we are seeing, a `DRAFT` post is being transformed into one with a `SCHEDULED` status. In the case of a post being private, if it's converted to `SCHEDULED` won't it lose the `PRIVATE` status which identifies it as such and thus having `isPublishDateInTheFuture` would work? 

Let me know what your thoughts are on this. I remember seeing a diagram once of the `PostStatuses` or was that something else because some of it is still a bit confusing for me 😅

Reply:
> I feel like we must have the exact some logic on the button in the editPostActivity, right? Could we perhaps re-use it. Wdyt? 

This is a good point! Thanks. Is the logic you are referring to this [EditorActionsProvider.kt](https://github.com/wordpress-mobile/WordPress-Android/blob/82f6294cc4c3587de6ba11d889aa3c72fe4046c4/WordPress/src/main/java/org/wordpress/android/ui/posts/editor/EditorActionsProvider.kt)

Reply:
> In the case of a post being private, if it's converted to SCHEDULED won't it lose the PRIVATE status which identifies it as such

The check starts with `initialPostStatus == DRAFT` so it's not a PRIVATE post, right? I'm not sure I understand the question.

>Let me know what your thoughts are on this. I remember seeing a diagram once of the PostStatuses or was that something else because some of it is still a bit confusing for me 😅

You might mean this diagram - paCBwp-95-p2

> Is the logic you are referring to this EditorActionsProvider.kt

yes

Reply:
> You might mean this diagram - paCBwp-95-p2 

Perfect! Exactly what I needed! 🙇‍♂️

Reply:
@malinajirka I deleted this class in the follow-up PR, so I want you to ignore it. I was going to just remove it from this PR but it would break things some tests so I just left it for now.